routing/vehicle_model.cpp

```diff
@@ -52,11 +62,16 @@ double VehicleModel::GetMinTypeSpeed(feature::TypesHolder const & types) const
     auto it = m_types.find(type);
     if (it != m_types.end())
       speed = min(speed, it->second);
+
+    auto const addRoadInfoIter = GetRoadTypeIter(type);
+    if (addRoadInfoIter != m_addRoadTypes.end())
+      speed = min(speed, addRoadInfoIter->m_speedKMpH);
   }
   if (speed <= m_maxSpeedKMpH)
     return speed;
 
-  return 0.0;
+  LOG(LERROR, ("Wrong routing types:", types));
+  return 0.5 /* Small speed to prevent routing along this edge. */;
```

Comment:
Why??? There can be types that don't suit to the routing. What's the purpose to route on them? Return zero as before.